### PR TITLE
Issues #19 and #20 - User can select what they want to see in the indoor map

### PR DESCRIPTION
**Pull request by Neeham**
#19 and #20:

User can now go under settings and select whether or not they would like to see the following options under indoor map: Stairs, Escalators, Elevators, Restrooms, Printers, Fountains, Fire Escape. It is important to note that a restriction has been added to prevent the user from deselecting all 3 of the following option: Stairs, Escalators and Elevators. The user must have at least one of the mentioned 3 options at all time.

TO DO under a separate branch: 
- Assign appropriate icon for each options shown under settings
- On deselecting options under settings, the indoor map is updated according to these setting preferences.

### CODE REVIEW

File: app/src/main/java/com/droidhats/campuscompass/views/CalendarFragment.kt [hunk 1]
```diff
@@ -151,6 +151,8 @@ class CalendarFragment : DialogFragment(), DialogInterface.OnDismissListener {
         calendarViewModel.selectedColors = checkedArr
         val sharedPreferences = activity?.getPreferences(Context.MODE_PRIVATE)
         val editor = sharedPreferences?.edit()
+
+
```
Comment (Amanda-Kolopa) [inline app/src/main/java/com/droidhats/campuscompass/views/CalendarFragment.kt:155]: Please remove the extra line.

Comment (Neeham) [reply app/src/main/java/com/droidhats/campuscompass/views/CalendarFragment.kt]: DONE! 👍

File: app/src/main/java/com/droidhats/campuscompass/views/SettingsFragment.kt [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+        val sharedPref = requireActivity().getPreferences(Context.MODE_PRIVATE)
+        val default: Boolean = sharedPref.getBoolean(buttonText, true)
+        switchButton.isChecked = default
+
+        if(restriction && switchButton.isChecked) {
+            totalCheck++
+        }
+    }
+
+
```
Comment (Amanda-Kolopa) [inline app/src/main/java/com/droidhats/campuscompass/views/SettingsFragment.kt:119]: Please remove the extra line.

File: app/src/main/java/com/droidhats/campuscompass/views/SettingsFragment.kt [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+    private fun populateSettingOffArray() {
+        val sharedPref = requireActivity().getPreferences(Context.MODE_PRIVATE)
+        val set = sharedPref.getStringSet("settingOffArray", null)
+
+        for (element in set!!) {
+            preferenceOff.add(element)
+        }
+    }
+
+
```
Comment (Amanda-Kolopa) [inline app/src/main/java/com/droidhats/campuscompass/views/SettingsFragment.kt:129]: Please remove the extra line.

File: app/src/main/res/values/colors.xml [hunk 4]
```diff
@@ -3,4 +3,5 @@
     <color name="colorPrimary">#008577</color>
     <color name="colorPrimaryDark">#980026</color>
     <color name="colorAccent">#D81B60</color>
+    <color name="lightGrey">#f2f2f2</color>
```
Comment (Amanda-Kolopa) [inline app/src/main/res/values/colors.xml:6]: Please remove this. There are no references to this color in the app.

File: app/src/main/java/com/droidhats/campuscompass/views/SettingsFragment.kt (hunk 2 above)
Comment (Neeham) [reply app/src/main/java/com/droidhats/campuscompass/views/SettingsFragment.kt:119]: Done 👍

File: app/src/main/res/values/colors.xml (hunk 4 above)
Comment (Neeham) [reply app/src/main/res/values/colors.xml:6]: Done 👍

File: app/src/main/java/com/droidhats/campuscompass/views/SettingsFragment.kt (hunk 3 above)
Comment (Neeham) [reply app/src/main/java/com/droidhats/campuscompass/views/SettingsFragment.kt:127]: Done 👍

**Comment by Neeham** [comment]
Thank you for your kind review @Amanda-Kolopa! The way you describe in details really helps big time to review and fix these issues 😄 

#### In terms of UI:
* We should add a title for the settings. For example, Indoor Navigation
-- Fixed 👍 
* We need to change the images (as discussed, this will be altered once all the images are available)
-- Yes this will be done in the future PR as the images are currently stored in a seperate branch. ✋ 
* Is it possible to move the Settings option to the bottom of the side menu?
-- TL;DR: It's not worth it but if really needed, we can use a hack/workaround which will not work properly on all devices using the following item:         
`<item android:name="paddingTop" android:title="">18dp</item>` 👎 

This one is a big issue. Nothing is impossible but that also brings us to adding lots of code for just this one change. The issue is that all those buttons are grouped are under the nav_drawer_main_menu.xml if we want to move the settings button to the bottom we will have to remove that item and put it under activity_main.xml (right after the portion of the code where the call is made to the menu). Now the issue with this is that, we now have to implement it's own onclick event listener in the code since it is now seperated from it's nav_drawer_main_menu.xml which to be honest is not worth it. However, there is a hack workaround this. We can keep everything the way it is and add another item right above the settings button under nav_drawer_main_menu.xml and that item will basically be for padding top the setting buttton. For this we need to specify the padding dp which will most likely cause an issue on different phone sizes.

#### In terms of Functionality:
* Works well! As discussed, there could be improvements by making more use of Preferences from Android Jetpack but is not a major issue.
-- If time permits then definitely will clean it up and improve it 😄 

#### In terms of UI tests:
6 out of 10 are failing when running them all together.

* CalendarFragmentTest → the test is selecting My Places instead of Schedule
-- Fixed 👍 
* NavigationDrawerTest (test_isSideMenuListElementsInView) → fails because it can't find the removed side menu option Resources.
-- fixed 👍 


* The rest is not related to this pull request but need to be adjusted:

DirectionsTest (test_SwitchToggle) → fails when checking for the toggle button. We need to clear the set navigation path from the previous tests for this to pass.
-- This is an easy fix but it requires us having the close button (to exist navigation) which I believe @KhaledBadran was working on it. The moment this button comes to existence we apply a click on it during test and it will be no longer fail ✋  

MarkerTest (testBuildingMarkers) → fails when trying to test the Loyola buildings. Same issue as before, we need to clear the previous navigation path.
-- Same situation as the previous test, awaiting for the close button ✋

**Comment by Amanda-Kolopa** [comment]
> Thank you for your kind review @Amanda-Kolopa! The way you describe in details really helps big time to review and fix these issues 
> 
> #### In terms of UI:
> * We should add a title for the settings. For example, Indoor Navigation
>   -- Fixed +1
> * We need to change the images (as discussed, this will be altered once all the images are available)
>   -- Yes this will be done in the future PR as the images are currently stored in a seperate branch. hand
> * Is it possible to move the Settings option to the bottom of the side menu?
>   -- TL;DR: It's not worth it but if really needed, we can use a hack/workaround which will not work properly on all devices using the following item:
>   `<item android:name="paddingTop" android:title="">18dp</item>` -1
> 
> This one is a big issue. Nothing is impossible but that also brings us to adding lots of code for just this one change. The issue is that all those buttons are grouped are under the nav_drawer_main_menu.xml if we want to move the settings button to the bottom we will have to remove that item and put it under activity_main.xml (right after the portion of the code where the call is made to the menu). Now the issue with this is that, we now have to implement it's own onclick event listener in the code since it is now seperated from it's nav_drawer_main_menu.xml which to be honest is not worth it. However, there is a hack workaround this. We can keep everything the way it is and add another item right above the settings button under nav_drawer_main_menu.xml and that item will basically be for padding top the setting buttton. For this we need to specify the padding dp which will most likely cause an issue on different phone sizes.
> 
> #### In terms of Functionality:
> * Works well! As discussed, there could be improvements by making more use of Preferences from Android Jetpack but is not a major issue.
>   -- If time permits then definitely will clean it up and improve it smile
> 
> #### In terms of UI tests:
> 6 out of 10 are failing when running them all together.
> 
> * CalendarFragmentTest → the test is selecting My Places instead of Schedule
>   -- Fixed +1
> * NavigationDrawerTest (test_isSideMenuListElementsInView) → fails because it can't find the removed side menu option Resources.
>   -- fixed +1
> * The rest is not related to this pull request but need to be adjusted:
> 
> DirectionsTest (test_SwitchToggle) → fails when checking for the toggle button. We need to clear the set navigation path from the previous tests for this to pass.
> -- This is an easy fix but it requires us having the close button (to exist navigation) which I believe @KhaledBadran was working on it. The moment this button comes to existence we apply a click on it during test and it will be no longer fail 
> 
> MarkerTest (testBuildingMarkers) → fails when trying to test the Loyola buildings. Same issue as before, we need to clear the previous navigation path.
> -- Same situation as the previous test, awaiting for the close button

Anytime @Neeham! Thank you for taking all the comments into consideration :smile: 

No worries about moving the location of the Settings side option. We can leave it where it is :slightly_smiling_face: 
I approve the changes!

**Pull request merged**